activerecord/lib/active_record/relation/where_clause.rb

```diff
@@ -107,9 +111,12 @@ def equality_node?(node)
         end
 
         def non_conflicting_binds(other)
-          conflicts = referenced_columns & other.referenced_columns
-          conflicts.map! { |node| node.name.to_s }
-          binds.reject { |attr| conflicts.include?(attr.name) }
+          # Choose all predicates that had bound params to work with them based on index
+          predicates_with_binds = predicates.select { |predicate| equality_node?(predicate) && predicate.right.is_a?(Arel::Nodes::BindParam) }
```

Comment:
@matthewd Thanks for looking at the PR. That's a good point. For some reason I thought that the Bind can only be on the right hand side. I will double check and do the necessary adjustments in the next few days.